File: LLM-Translations/GPT-4/codenet/Java/Go/s493101281.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
	"strconv"
)

const MOD int64 = 1000000007

func main() {
	sc := bufio.NewScanner(os.Stdin)
	sc.Split(bufio.ScanWords)

	n := nextInt(sc)
	l := make([]int, n)

	for i := 0; i < n; i++ {
		l[i] = nextInt(sc)
	}

	sort.Ints(l)

	fmt.Println(l[n/2] - l[n/2-1])
}

func nextInt(sc *bufio.Scanner) int {
	sc.Scan()
	i, err := strconv.Atoi(sc.Text())
	if err != nil {
		panic(err)
	}
	return i
}

func gcd(m, n int64) int64 {
	if m < n {
		return gcd(n, m)
	}
	if n == 0 {
		return m
	}
	return gcd(n, m%n)
}

func lcm(m, n int64) int64 {
	return m * n / gcd(m, n)
}

type Pair struct {
	key   int
	value int
}

func (p *Pair) getValue() int {
	return p.value
}

func (p *Pair) getKey() int {
	return p.key
}

// 

File: LLM-Translations/GPT-4/codenet/Java/Python/s493101281.py
import sys
from typing import List, Tuple

def main():
    MOD = 1000000007

    def input(): return sys.stdin.readline().strip()
    def iinput(): return int(input())
    def linput(): return list(map(int, input().split()))
    def lcm(m: int, n: int) -> int: return m * n // gcd(m, n)
    def gcd(m: int, n: int) -> int: return m if n == 0 else gcd(n, m % n)

    n = iinput()
    l = linput()
    l.sort()

    print(l[n//2] - l[n//2 - 1])

if __name__ == "__main__":
    main()


